File: 2018 - 1/Estruturas de Dados/Alguns códigos/EstruturasDeDadosEmC/Fila.c
#include "Fila.h"
#include <stdio.h>
#include <stdbool.h>




        void Cria(struct Fila *f){
          f->primeiro = 0;
          f->ultimo = 0;
          f->total = 0;

      }
   void Insere(struct Fila *f, int elem){
            f->elementos[f->ultimo] = elem;
            f->ultimo = (f->ultimo+1) % 10;
            f-> total = f->total + 1;
            printf("Inserindo %d \n", elem );
    }
     int Retira(struct Fila *f ){
        int elem = f->elementos[f->primeiro];
        f->primeiro = (f->primeiro +1 ) %10;
            f-> total = f->total - 1;
        return elem;

    }
     int Cheia(struct Fila f){
        return (f.total == 10);

    }
     int Vazia(struct Fila f){
            return (f.total == 0);
}

File: 2018 - 1/Estruturas de Dados/Alguns códigos/Fila.cs
struct Node{
	char Info;
	NodePtr Next;


};

struct Fila{
	NodePtr inicio;
	NodePtr final;  // com fila vazia, ambos apontam pra null;
//	int qtdElementos;

};

public void Insere(Fila *f, char x, boolean DeuCerto){
if(Cheia(f) == true)
	DeuCerto = false;
	else{
		paux = newNode();
		paux->Next = null;
		final-> Next = paux;
		final = paux;

	}

}
public void Retira(Fila *f, char x, boolean DeuCerto){
		if(Vazia(f)
			DeuCerto = false;
			else{
				
					paux = f.inicio;
					f.inicio = f.Next;
					deleteNode(paux);
}
}
public boolean Cheia(Fila *f){
	return false;

}
public boolean Vazia(Fila *f){
	if(f.inicio == null && f.final == null)
	return true;
	else{return false;}

}
public void Cria(Fila *f){
		f.inicio  = null;
		  f.final = null;


}